docs/specs/tsd-009-probe-on-startup.md

```diff
@@ -0,0 +1,68 @@
+# Probing plugins after startup
+
+## Objective
+
+Allow Telegraf to probe plugins during startup to enable enhanced plugin error
+detection like availability of hardware or services
+
+## Keywords
+
+inputs, outputs, startup, probe, error, ignore, behavior
+
+## Overview
+
+When plugins are first instantiated, Telegraf will call the plugin's `Start()`
+method (for inputs) or `Connect()` (for outputs) which will initialize its
+configuration based off of config options and the running environment. It is
+sometimes the case that while the initialization step succeeds, the upstream
+service in which the plugin relies on is not actually running, or is not capable
+of being communicated with due to incorrect configuration or environmental
+problems. In situations like this, Telegraf does not detect that the plugin's
```

Comment:
Initialisation happens in `Init()`, connection attempts in `Start()`/`Connect()`.

Reply:
@LandonTClipp